tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -0,0 +1,205 @@
+assert = chai.assert
+
+class FakeHighlight
+
+  constructor: (@segment, @anchor, @page) ->
+
+  @createFrom: (segment, anchor, page) ->
+    new FakeHighlight segment, anchor, page
+
+# Since the system for registering highlight types have been obliterated,
+# for now, we will do a hard override of the TextHighlight class
+Annotator.TextHighlight = FakeHighlight
+
+class FakeAnchor extends Annotator.Anchor
+
+  _getSegment: -> "html segment for " + @id
+
+  constructor: (manager, annotation, target) ->
+    super manager, annotation, target, 0, 0,
+      "fake quote for" + target.id
+
+    @id = "fake anchor for " + target.id
```

Comment:
Again just stubing `Anchor` and returning the object you want is clearer than subclassing and monkey patching the object in the test.